bot/exts/evergreen/wikipedia.py

```diff
@@ -46,37 +47,39 @@ async def wiki_request(self, channel: TextChannel, search: str) -> Optional[List
         """Search wikipedia search string and return formatted first 10 pages found."""
         params = WIKI_PARAMS | {"srlimit": 10, "srsearch": search}
         async with self.bot.http_session.get(url=SEARCH_API, params=params) as resp:
-            if resp.status == 200:
-                raw_data = await resp.json()
-                number_of_results = raw_data["query"]["searchinfo"]["totalhits"]
-
-                if number_of_results:
-                    results = raw_data["query"]["search"]
-                    lines = []
-
-                    for article in results:
-                        line = WIKI_SEARCH_RESULT.format(
-                            name=article["title"],
-                            description=unescape(
-                                re.sub(
-                                    WIKI_SNIPPET_REGEX, "", article["snippet"]
-                                )
-                            ),
-                            url=f"https://en.wikipedia.org/?curid={article['pageid']}"
-                        )
-                        lines.append(line)
-
-                    return lines
-
-                else:
-                    await channel.send(
-                        "Sorry, we could not find a wikipedia article using that search term."
+            if (status := resp.status) != 200:
+                log.info(f"Unexpected response `{status}` while searching wikipedia for `{search}`")
+                raise ExternalAPIError("Wikipedia API", status)
+
+            raw_data = await resp.json()
+            if not raw_data.get("query"):
+                if error := raw_data.get("errors"):
+                    log.error(f"There was an error while communicating with the Wikipedia API: {error}")
+                raise ExternalAPIError("Wikipedia API", status, error)
+
+            number_of_results = raw_data["query"]["searchinfo"]["totalhits"]
+
+            if number_of_results:
+                results = raw_data["query"]["search"]
+                lines = []
+
+                for article in results:
+                    line = WIKI_SEARCH_RESULT.format(
+                        name=article["title"],
+                        description=unescape(
+                            re.sub(
+                                WIKI_SNIPPET_REGEX, "", article["snippet"]
+                            )
+                        ),
+                        url=f"https://en.wikipedia.org/?curid={article['pageid']}"
                     )
-                    return
+                    lines.append(line)
+
+                return lines
+
             else:
-                log.info(f"Unexpected response `{resp.status}` while searching wikipedia for `{search}`")
                 await channel.send(
-                    "Whoops, the Wikipedia API is having some issues right now. Try again later."
+                    "Sorry, we could not find a wikipedia article using that search term."
                 )
                 return
```

Comment:
Passing an empty list here instead of having the else block here would be better as you are already checking if `contents` is populated in the `wikipedia_search_command`. 
```suggestion
            return lines
```
The else block can be used in the `wikipedia_search_command`. This will require you to define the `lines` variable (line 64) outside the if-else scope.
And also don't forget to change the function return annotation to only `List[str]` instead of `Optional[List[str]]`.